CMakeLists.txt

```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.9)
```

Comment:
Why is this necessary? Default on Ubuntu 16.04 is 3.5.1 so we can't go higher than that.

Reply:
I have checked the documentation of CMake and put the reasons in the commit comment:

In "dali/CMakeLists.txt":44
CUDA_LINK_LIBRARIES_KEYWORD is available in CMake 3.9 or above

In "dali/pipeline/operators/reader/parser/CMakeLists.txt":22
Parameter "EXPORT_MACRO" for protobuf_generate_cpp is available in CMake 3.9 or above

Reply:
I have checked the documentation of CMake and put reasons in the comment of this commit.

In "dali/CMakeLists.txt":44
CUDA_LINK_LIBRARIES_KEYWORD is available in CMake 3.9 or above

In "dali/pipeline/operators/reader/parser/CMakeLists.txt":22
Parameter "EXPORT_MACRO" for protobuf_generate_cpp is available in CMake 3.9 or above

Reply:
For CUDA_LINK_LIBRARIES_KEYWORD, there is a workaround:
set(CUDA_LIBRARIES ${CUDA_LINK_LIBRARIES_KEYWORD} ${CUDA_LIBRARIES})
applying this when cmake is below 3.9

As for EXPORT_MACRO" parameter of protobuf_generate_cpp, I haven't found any solution for CMake below 3.9 for now